devicetypes/smartthings/zwave-multi-metering-switch.src/zwave-multi-metering-switch.groovy

```diff
@@ -213,71 +246,85 @@ def ping() {
 	refresh()
 }
 
-def reset(endpoint = 1) {
-	log.debug "Resetting endpoint: ${endpoint}"
-	delayBetween([
-			secureEncap(zwave.meterV3.meterReset(), endpoint),
-			secureEncap(zwave.meterV3.meterGet(scale: 0), endpoint),
-			"delay 500"
-	], 500)
-}
-
 def childOnOff(deviceNetworkId, value) {
 	def switchId = getSwitchId(deviceNetworkId)
 	if (switchId != null) sendHubCommand onOffCmd(value, switchId)
 }
 
-def childRefresh(deviceNetworkId) {
-	def switchId = getSwitchId(deviceNetworkId)
-	if (switchId != null) sendHubCommand refresh(switchId)
+private onOffCmd(value, endpoint = 1) {
+	delayBetween([
+			secureEncap(zwave.basicV1.basicSet(value: value), endpoint),
```

Comment:
All the messages should not by default go through `secureEncap`, they should go through another method that determines if secureEncapsulation is necessary.

Reply:
Perhaps `secureEncap` is doing exactly what I am asking for, but the naming of that function is not appropriate.

Reply:
Yes, secureEncap() was doing it, but - because of that naming - i have refactored that code and now everything is in encap(...) method